Tests/Common/SonarQube/TestLogger.cs

```diff
@@ -26,20 +26,16 @@ public bool IncludeTimestamp
 
         public TestLogger()
         {
-            // Write out a separator. Many tests create more than one TestLogger.
-            // This helps separate the results of the different cases.
-            WriteLine("");
-            WriteLine("------------------------------------------------------------- (new TestLogger created)");
-            WriteLine("");
-
-            DebugMessages = new List<string>();
-            InfoMessages = new List<string>();
-            Warnings = new List<string>();
-            Errors = new List<string>();
+            this.DoReset("------------------------------------------------------------- (new TestLogger created)");
         }
 
         #region Public methods
 
+        public void Reset()
+        {
+            this.DoReset("------------------------------------------------------------- (TestLogger reset)");
+        }
+
```

Comment:
Miscellaneous: added the ability to reset a test logger to simplify checks in tests that cover multiple cases (means the test can reset the logger and then call e.g. *logger.AssertErrorsLogged(0)*.